snyk/snyk-mvn-plugin
# fix: handle sha-not-found catalog error

#### What does this PR do?
The error coming back for the SHA1 query has changed its structure to be an ErrorCatalog error. 
This PR: 
* fixes the logic so that it can handle and errorCatalog type error without failing by itself.
* Jestifies the jar tests and adds a test to verify the change (`inspect on user created jar marks package as unknown`)


New response from a 404 (unknown) SHA1 query:
```
{
  "jsonapi": {
    "version": "1.0"
  },
  "errors": [
    {
      "id": "117b62de-1678-419f-a995-5860e408dc8b",
      "links": {
        "about": "https://docs.snyk.io/scan-with-snyk/error-catalog#snyk-osjvm-002"
      },
      "status": "404",
      "code": "SNYK-OSJVM-002",
      "title": "SHA1 not found",
      "detail": "SHA1 450ec19fa69e51c61335f431ee8df8657876e970 was not found",
      "meta": {
        "links": [
          "https://docs.snyk.io/snyk-cli/test-for-vulnerabilities/scan-all-unmanaged-jar-files"
        ],
        "sha1": "450ec19fa69e51c61335f431ee8df8657876e970",
        "isErrorCatalogError": true,
        "classification": "ACTIONABLE"
      }
    }
  ]
}

```
#### Where should the reviewer start?


#### How should this be manually tested?
If you run the current CLI on a made up jar file (`echo 1234 > fake.jar && snyk test --file=fake.jar -d`) you will see an error `INFO: Exiting on EOFFailed to get maven dependency for 'location/of/fake.jar'. Cannot read properties of undefined (reading 'map')` 

if you build and link this branch to the cli and run `npx ts-node /location/of/snyk/cli/src/cli/index.ts test --file=fake.jar -d` - the catalog error will be handled.

#### Any background context you want to provide?
[DD](https://app.datadoghq.com/logs?query=@err.name:Sha1NotFoundError%20&agg_m=count&agg_m_source=base&agg_t=count&cols=host,service,@requestUrl,@status,@err.metadata.status,@err.additionalData.sha1&fromUser=true&messageDisplay=inline&storage=hot&stream_sort=desc&viz=stream&from_ts=1729324027295&to_ts=1729496827295&live=true)

#### What are the relevant tickets?


#### Screenshots


#### Additional questions
